demos/imagenet_classification_demo/main.cpp

```diff
@@ -0,0 +1,503 @@
+        std::queue<InferRequestInfo> completedInferRequests;
+        for (std::size_t i = 0; i < inferRequests.size(); i++) {
+            emptyInferRequests.push({inferRequests[i], std::vector<InferRequestInfo::InferRequestImage>()});
+        }
+
+        auto startTime = std::chrono::steady_clock::now();
+        auto elapsedSeconds = std::chrono::steady_clock::duration{0};
+        // ---------------------------------------------------------------------------------------------------
+        
+        // -------------------------------------Processing infer requests-------------------------------------
+        do {
+            if (isTestMode && elapsedSeconds >= std::chrono::seconds{3}) {
+                isTestMode = false;
+                gridMat = GridMat(presenter, cv::Size(width, height), cv::Size(16, 9), avgFPS);
+                startTickCount = std::chrono::steady_clock::now();
+                framesNum = 0;
+                latencySum = 0;
+            }
+
+            std::unique_lock<std::mutex> lock(mutex);
+            if (!completedInferRequests.empty()) {
+                auto completedInferRequestInfo = completedInferRequests.front();
+                completedInferRequests.pop();
+                lock.unlock();
+                emptyInferRequests.push({completedInferRequestInfo.inferRequest, 
+                                         std::vector<InferRequestInfo::InferRequestImage>()});
+
+                std::vector<unsigned> rightClasses = {};
+                for (size_t i = 0; i < completedInferRequestInfo.images.size(); i++) {
+                    rightClasses.push_back(completedInferRequestInfo.images[i].rightClass);
+                }
+
+                std::vector<std::vector<unsigned>> results = topResults(
+                    *completedInferRequestInfo.inferRequest.GetBlob(outputName), FLAGS_nt);
+                std::vector<std::string> predictedLabels = {};
+                for (size_t i = 0; i < FLAGS_b; i++) {
+                    PredictionResult predictionResult = PredictionResult::Incorrect;
+                    if (!FLAGS_gt.empty()) {
+                        for (size_t j = 0; j < FLAGS_nt; j++) {
+                            unsigned predictedClass = results[i][j];
+                            if (predictedClass == rightClasses[i]) {
+                                predictionResult = PredictionResult::Correct;
+                                predictedLabels.push_back(labels[predictedClass]);
+                                correctPredictionsCount++;
+                                break;
+                            }
+                        }
+                    } else {
+                        predictionResult = PredictionResult::Unknown;
+                    }
+
+                    if (predictionResult != PredictionResult::Correct) {
+                        predictedLabels.push_back(labels[results[i][0]]);
+                    }
+                    totalPredictionsCount++;
```

Comment:
I think `totalPredictionsCount` is equal to `framesNum`. One of vars can be removed.